src/Matcher.js

```diff
@@ -0,0 +1,72 @@
+const { getData } = require('./Fetcher');
+const { writeJson, readJson } = require('./Json');
+const { crawlDataSkin } = require('./Crawler');
+const { credentials } = require('../config.json');
+const log = require('node-pretty-log');
+
+
+const getUsedWidgets = slots => {
+    let widgets = [];
+    Object.values(slots).forEach(slot => {
+        widgets = [...widgets, ...slot.widgets || []];
+    });
+    return widgets;
+};
+
+const crawlThroughInfo = (dataSkins, centralInfo) => {
+    centralInfo.forEach(centralConfig => {
+        const widgets = getUsedWidgets(centralConfig.slots);
+        widgets.forEach(widget => {
+            const widgetObject = centralConfig.widgets[widget];
+            const specifiedWidget = dataSkins[widgetObject.feature];
+            const widgetSkin = widgetObject.skin || 'default';
+            const isWidgetEnabled = widgetObject.enabled;
+            if (specifiedWidget[widgetSkin]) {
+                specifiedWidget[widgetSkin]
+                    .enabled = specifiedWidget[widgetSkin].enabled || isWidgetEnabled;
+            } else {
+                specifiedWidget.notListed = specifiedWidget.notListed || {};
+                specifiedWidget.notListed[widgetSkin] = [
+                    ...(specifiedWidget.notListed[widgetSkin] || []),
+                    { location: centralConfig.config.name, enabled: isWidgetEnabled },
+                ];
+            }
+        });
+    });
+    return dataSkins;
+};
+
+const getCrawledData = apiKey => {
+    const dataPath = `./objects/${apiKey}.json`;
+
+    return new Promise(resolve => {
+        readJson(dataPath)
+            .then(data => resolve(data))
+            .catch(() => resolve(crawlDataSkin(apiKey)));
+    });
+};
+
+module.exports = {
+    matchInfo: apiKey => {
+        const centralUrl = `https://vitrines-staging.chaordic.com.br/api/${apiKey}/templates`;
+        const centralInfo = getData(centralUrl, credentials)
+            .then(response => response.json())
+            .then(data => {
```

Comment:
You could use just one `then` putting `data = response.json()` on the first line of the block

Reply:
But I would add a line anyway because I would need to assign a variable with the result of response.json() to remain DRY